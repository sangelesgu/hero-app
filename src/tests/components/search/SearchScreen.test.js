import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';
import { SearchScreen } from '../../../components/search/SearchScreen';

describe('Test in <SearchScreen/>', () => {
  
  test('should display corretly with default values', () => {
    const wrapper = mount (
      <MemoryRouter initialEntries={['/search']}>
        <Route path="/search" component={SearchScreen}/>
      </MemoryRouter>
    );    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.alert-info').text().trim()).toBe('Search a hero..')
  }); 

  test('should show batman and input with value of queryString', () => {
    const wrapper = mount (
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <Route path="/search" component={SearchScreen}/>
      </MemoryRouter>
    );
    
    expect(wrapper.find('input').prop('value')).toBe('batman');
    expect(wrapper).toMatchSnapshot();
  })
  
  test('should return error if not find hero', () => {
    const wrapper = mount (
      <MemoryRouter initialEntries={['/search?q=batman123131']}>
        <Route path="/search" component={SearchScreen}/>
      </MemoryRouter>
    );
    expect(wrapper.find('.alert-danger').text().trim()).toBe('There is not a hero with batman123131');
    expect(wrapper).toMatchSnapshot();
  }); 

  test('should call push of history', () => {
    const history = {
      push: jest.fn()
    }

    const wrapper = mount (
      <MemoryRouter initialEntries={['/search?q=batman123131']}>
        <Route 
          path="/search" 
          component={ () => <SearchScreen history={history}/>}
        />
      </MemoryRouter>
    );

    wrapper.find('input').simulate('change', {
      target: {
        name: 'searchText', 
        value: 'batman'
      }
    }); 

    wrapper.find('form').prop('onSubmit')({
      preventDefault(){}
    }); 

    expect(history.push).toHaveBeenCalledWith('?q=batman');
  })
  
  
  
})
