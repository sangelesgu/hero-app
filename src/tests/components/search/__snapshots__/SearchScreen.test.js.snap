// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SearchScreen/> should display corretly with default values 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search a hero..
      </div>
    </div>
  </div>
</div>
`;

exports[`Test in <SearchScreen/> should return error if not find hero 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman123131"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="alert alert-danger"
      >
        There is not a hero with 
        batman123131
      </div>
    </div>
  </div>
</div>
`;

exports[`Test in <SearchScreen/> should show batman and input with value of queryString 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman"
        />
        <button
          className="btn m-1 btn-block btn-outline-primary"
          type="submit"
        >
          Search...
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <a
        className="my-card"
        href="/hero/dc-batman"
        onClick={[Function]}
      >
        <img
          alt="Batman"
          className="img img-responsive"
          src="./assets/heroes/dc-batman.jpg"
        />
        <div
          className="profile-name"
        >
          Batman
        </div>
        <div
          className="profile-position"
        >
          Bruce Wayne
        </div>
        <div
          className="profile-overview"
        >
          <div
            className="profile-overview"
          >
            <div
              className="row"
            >
              <div
                className="col-ms-4"
              >
                <h3>
                  DC Comics
                </h3>
                <p>
                  Primera aparici√≥n: 
                  <br />
                  Detective Comics #27
                </p>
              </div>
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
